src/pages/index.js

```diff
@@ -0,0 +1,329 @@
+
+    cards.forEach((item) => {
+      const cardElement = getCardElement(item);
+      cardsList.append(cardElement);
+    });
+  })
+  .catch(console.error);
+
+// Profile elements
+const profileEditButton = document.querySelector(".profile__edit-button");
+const cardModalButton = document.querySelector(".profile__add-button");
+const avatarModalButton = document.querySelector(".profile__avatar-button");
+const profileName = document.querySelector(".profile__name");
+const profileDescription = document.querySelector(".profile__description");
+const profileEditModal = document.querySelector("#profile-edit-modal");
+const profileFormElement = profileEditModal.querySelector(".modal__form");
+const editModalCloseButton = profileEditModal.querySelector(
+  ".modal__close-button"
+);
+const editModalNameInput = profileEditModal.querySelector(
+  "#profile-name-input"
+);
+const editModalDescriptionInput = profileEditModal.querySelector(
+  "#profile-description-input"
+);
+
+// Card elements
+const cardModal = document.querySelector("#add-card-modal");
+const cardForm = cardModal.querySelector(".modal__form");
+const cardSubmitButton = cardModal.querySelector(".modal__submit-button");
+const cardModalCloseButton = cardModal.querySelector(".modal__close-button");
+const cardNameInput = cardModal.querySelector("#add-card-name-input");
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+// Avatar elements
+const avatarModal = document.querySelector("#profile-avatar-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarSubmitButton = avatarModal.querySelector(".modal__submit-button");
+const avatarModalCloseButton = avatarModal.querySelector(
+  ".modal__close-button"
+);
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+
+// Delete elements
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form");
+
+// Preview elements
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseButton = previewModal.querySelector(
+  ".modal__close-button"
+);
+
+let selectedCard;
+let selectedCardId;
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = evt.submitter;
+  submitButton.disabled = true;
+  setButtonText(submitButton, true);
+  api
+    .getNewCard({ name: cardNameInput.value, link: cardLinkInput.value })
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      cardForm.reset();
+      resetValidation(cardForm, settings);
+      closeModal(cardModal);
+    })
+    .catch(console.error, (submitButton.disabled = false))
```

Comment:
`    .catch(console.error, (submitButton.disabled = false))
`

It doesn't work like that because the button gets activated  immediately


It should be this way:

```js
    .catch((err) => {
      console.error(err);
      submitButton.disabled = false;
    })

```